controllers/keda/scaledobject_controller_test.go

```diff
@@ -221,6 +220,7 @@ var _ = Describe("ScaledObjectController", func() {
 
 				// Call function tobe tested
 				metricSpecs, err := metricNameTestReconciler.getScaledObjectMetricSpecs(context.Background(), testLogger, duplicateNamedScaledObject)
+				Ω(err).ShouldNot(BeNil())
```

Comment:
I added this assertion and remove one of the test case above because of this error 
```
• [FAILED] [0.013 seconds]
ScaledObjectController Metric Names With Unique Values [It] should pass metric name validation
/home/tung/go/src/keda/controllers/keda/scaledobject_controller_test.go:90

  Timeline >>
  2023-08-05T10:49:11+08:00	INFO	starting server	{"path": "/metrics", "kind": "metrics", "addr": "[::]:8080"}
  2023-08-05T10:49:11+08:00	INFO	Starting EventSource	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "source": "kind source: *v1alpha1.ScaledObject"}
  2023-08-05T10:49:11+08:00	INFO	Starting EventSource	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject", "source": "kind source: *v2.HorizontalPodAutoscaler"}
  2023-08-05T10:49:11+08:00	INFO	Starting Controller	{"controller": "scaledobject", "controllerGroup": "keda.sh", "controllerKind": "ScaledObject"}
  2023-08-05T10:49:11+08:00	DEBUG	certificates	the path /custom/ca doesn't exist, skipping custom CA registrations
  [FAILED] in [It] - /home/tung/go/src/keda/controllers/keda/scaledobject_controller_test.go:133 @ 08/05/23 10:49:11.747
  << Timeline

  [FAILED] Expected
      <*errors.errorString | 0xc001c73aa0>: 
      metricName s0-prometheus-prometheus defined multiple times in ScaledObject , please refer the documentation how to define metricName manually
      {
          s: "metricName s0-prometheus-prometheus defined multiple times in ScaledObject , please refer the documentation how to define metricName manually",
      }
```

Reply:
That test case shouldn't fail because `sX-` should be unique within ScaledObject, let me check the code to see if there is any issue

Reply:
I got the reason 😄 
It's because during the iteration, the parameter `ScalerIndex` isn't passed. As now we don't override the metricName, both have the index 0 with the same name.
Replace the line 94 with:
```golang
config := &scalers.ScalerConfig{
	ScalableObjectName:      fmt.Sprintf("test.%d", i),
	ScalableObjectNamespace: "test",
	TriggerMetadata:         tm,
	ResolvedEnv:             nil,
	AuthParams:              nil,
	ScalerIndex:             i,
}
```
Adding the line `ScalerIndex: i` you will generate the metric names correctly and it will work (as it should work indeed)